community/community.cpp

```diff
@@ -495,6 +508,20 @@ void cambiatus::claimaction(std::uint64_t action_id, eosio::name maker)
   // Check if the action is claimable
   eosio::check(objact.verification_type == "claimable", "You can only open claims in claimable actions");
 
+  // Check action proofs
+  eosio::check(proof_photo.length() <= 256,
+               "Invalid length for proof photo url, must be less than 256 characters");
+  eosio::check(proof_code.length() <= 30,
+               "Invalid length for proof code, must be less than 30 characters");
```

Comment:
@newdev42 shouldn't those validations be inside the `if`s below?

Reply:
not necessarily, imagine the user submitted a proof photo and a proof code that's bigger than what is expected by mistake, regardless of the the `has_proof_photo` and `has_proof_code` flags on the action table. actually, it's a good point, should we reject it right away if the flags are not true? (maybe we want to allow people to attach photos even if the actions don't require it?)

Reply:
On the other actions, we reject everything right away! I think its easier to debug and to teach users how to use it too!

Reply:
for the proof_photo: so leaving it as-is (outside the following ifs) will reject it right away. is that what you meant? just leave it? 

(moved the proof code check to it's own validation prior the hash)

Reply:
done!